api/src/main/java/org/openmrs/api/ObsService.java

```diff
@@ -537,4 +539,51 @@ public Integer getObservationCount(List<Person> whom, List<Encounter> encounters
 			Integer obsGroupId, Date fromDate, Date toDate, boolean includeVoidedObs, String accessionNumber)
 			throws APIException;
 	
+
+	/**
+	 * Creates or updates the given {@link ObsReferenceRange} in the database
+	 *
+	 * @param obsReferenceRange the obsReferenceRange to save
+	 *                             
+	 * @since 2.7.0
+	 * 
+	 * <strong>Should</strong> create a new obs reference range
+	 * 	 * <strong>Should</strong> edit an existing obs reference range
+	 */
+	ObsReferenceRange saveObsReferenceRange(final ObsReferenceRange obsReferenceRange);
+
+	/**
+	 * @param id ObsReferenceRange id
+	 * @return {@link ObsReferenceRange}
+	 *
+	 * @since 2.7.0
+	 *
+	 * <strong>Should</strong> get the ObsReferenceRange with the given id
+	 * <strong>Should</strong> return null if ObsReferenceRange is not found
+	 */
+	@Authorized(PrivilegeConstants.GET_OBS)
+	ObsReferenceRange getObsReferenceRangeById(final Integer id);
+
+	/**
+	 * @param obsId observation id 
+	 * @return list of {@link ObsReferenceRange}
+	 *
+	 * @since 2.7.0
+	 *
+	 * <strong>Should</strong> get a list of ObsReferenceRange with the given obsId
+	 * <strong>Should</strong> return empty list if none of ObsReferenceRanges has the given obsId
+	 */
+	@Authorized(PrivilegeConstants.GET_OBS)
+	List<ObsReferenceRange> getObsReferenceRangesByObsId(final Integer obsId);
```

Comment:
I think we are going to have a maximum of one reference range for an observation. Not so? https://www.hl7.org/fhir/observation-definitions.html#Observation.referenceRange

Reply:
According to FHIR, the cardinality is 1...* (meaning reference ranges can be > 1), so it can be a list. But I don't mind if we have to tweak something around it. Any other ideas?

Reply:
This makes a lot sense. Let's use 1:1 relationship